cdk-testsuite/tests/vagrant_sanity.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/python
+
+from avocado import Test
+from avocado.utils import process
+import os, re, time, pexpect, vagrant
+
+class VagrantSanity(Test):
+    def setUp(self):
+	self.vagrant_VAGRANTFILE_DIR = self.params.get('vagrant_VAGRANTFILE_DIR')
+        self.vagrant_PROVIDER = self.params.get('vagrant_PROVIDER', default='')
+	self.vagrant_RHN_USERNAME = self.params.get('vagrant_RHN_USERNAME')
+	self.vagrant_RHN_PASSWORD = self.params.get('vagrant_RHN_PASSWORD')
+	os.chdir(self.vagrant_VAGRANTFILE_DIR)
+	self.v = vagrant.Vagrant(self.vagrant_VAGRANTFILE_DIR)
+
+
+    def vagrant_status(self):
+    	''' checks status of vagrant box and returns the state '''
+        self.log.info("Checking the status of the vagrant box...")
+        out = self.v.status()
+        state = re.search(r"state='(.*)',", str(out) ).group(1)
+        self.log.info("State of the vagrant box is %s" %(state))
+        return state
+
+
+    def test_vagrant_up(self):
+    	''' destroy old box and bring up a the vagrant box and finally destroy '''
+	self.log.info("Destroying old vagrant box, if any...")
+	self.test_vagrant_destroy()
+	#cmd = "vagrant up --provider %s" %(self.vagrant_PROVIDER)
+	self.log.info("Brining up the vagrant box...")
+	child = pexpect.spawn ('vagrant up')
+	child.expect('.*Would you like to register the system now.*', timeout=300)
```

Comment:
How it behaves when there is no propt for user credentials?
All these use cases should be supported by this test:
1) no registration plugin installed (adb testing)
2) credentials in environment variables (no prompt)
3) Vagrantfile without registration plugin used